### DISCUSSION TITLE
Node blueprint folder structure

### DISCUSSION THREAD

**Discussion by thomashdk** [umbraco/Umbraco-CMS#15565] [Features and ideas]
Some of our clients have a lot of blueprints for a specific doctype.

It could be nice the be able to create folder to orginize them a litte bit more.

I would love to do it, but i'm not sure how to tab in to that view. Maybe if i'm pointed in a direction I can take a look again :)

![image](https://github.com/umbraco/Umbraco-CMS/assets/6445636/bc7d8a0d-d0e6-4a70-9254-6ca4c7037726)

**Comment by bergmania** [comment]
Hi @thomashdk 

The blueprints structure is not really a normal tree structure.

I'm bring your request to an internal meeting, where we have to discus how to implement that tree for v14 anyway. Maybe we can come up with a better solution in the current.

**Reply by thomashdk** [reply]
Thanks 🙌🏻 😀

**Reply by ronaldbarendse** [reply]
Document blueprint containers (folders) have been implemented in v14, so you can now create multiple hierarchical levels to organize them a bit more, see: https://github.com/umbraco/Umbraco-CMS/pull/15983.

The C# API requires some more work though, as `IContentBlueprintContainerService` does not have a way to get all items, only root items, children and/or descendants (only get the entity by key and get the parent). This can be partially be done using `IEntityService`, but some methods on this interface are not supported for entity containers, e.g. `IEntityService.GetAll(Constants.ObjectTypes.DocumentBlueprintContainer)` throws a NotSupportedException` 😢
